lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -1248,20 +1247,30 @@ static void copyParameterArgumentsForApply(
 /// recursively converts the new function just like how the old function is
 /// converted. If the new function's generic signature is specified, it is used
 /// to create substitution maps for reapplied `partial_apply` instructions.
-static SILValue
-reapplyFunctionConversion(
-    SILValue newFunc, SILValue oldFunc, SILValue oldConvertedFunc,
-    SILBuilder &builder, SILLocation loc,
+static SILValue reapplyFunctionConversion(
+    ADContext &context, SILValue newFunc, SILValue oldFunc,
+    SILValue oldConvertedFunc, SILBuilder &builder, SILLocation loc,
     SmallVectorImpl<AllocStackInst *> &newBuffersToDealloc,
-    GenericSignature newFuncGenSig = GenericSignature()) {
+    GenericSignature newFuncGenSig = GenericSignature(),
+    IndexSubset *reabstractionThunkIndices = nullptr) {
```

Comment:
Is there a reason why `reabstractionThunkIndices` is only set when dealing with reabstraction thunks? Would it make sense to make it a more general thing that we require callers to set in all cases?

I can't think of any reason why we'd need the information in other cases, but it would simplify the API and make mistakes where callers don't set the indices less likely.

Reply:
Thanks for the suggestion!

https://github.com/apple/swift/pull/28570/commits/6eb7a070fd6734c1deb1220556038e4267d349ef changes `reapplyFunctionConversion` so that parameter indices are always passed. Reabstraction thunk special logic is triggered when the original callee is a reabstraction thunk `function_ref` - this is a nicer condition.